interp/ast.go

```diff
@@ -565,7 +569,14 @@ func (interp *Interpreter) ast(src, name string) (string, *node, error) {
 			st.push(addChild(&root, anc, pos, caseClause, aCase), nod)
 
 		case *ast.ChanType:
-			st.push(addChild(&root, anc, pos, chanType, aNop), nod)
+			switch a.Dir {
+			case ast.SEND | ast.RECV:
```

Comment:
nitpick: shouldnt this be the default case. It might be better to try work out why it is bidirectional than to work out why it isnt added.

Reply:
Would it be possible to have chan that is neither send or receive ? :thinking:

Reply:
I dont think so, not much of a channel if it cannot send or receive.